lib/EnzymeCore/src/rules.jl

```diff
@@ -209,35 +209,15 @@ function isapplicable(@nospecialize(f), @nospecialize(TT);
     end
     fullmatch = Core.Compiler._any(match::Core.MethodMatch->match.fully_covers, matches)
     if !fullmatch
-        if caller isa Core.MethodInstance
-            add_mt_backedge!(caller, mt, sig)
-        elseif caller isa Core.Compiler.MethodLookupResult
-            for j = 1:Core.Compiler.length(caller)
-                cmatch = Core.Compiler.getindex(caller, j)::Core.MethodMatch
-                cspec = Core.Compiler.specialize_method(cmatch)::Core.MethodInstance
-                add_mt_backedge!(cspec, mt, sig)
-            end
-        end
+        add_mt_backedge!(caller, mt, sig)
```

Comment:
you still need the isa MethodInstance here and below since nothing could be passed